stdlib/asm/crypto/fri.masm

```diff
@@ -0,0 +1,594 @@
+    exec.accumulate_for_even_index
+
+    # for j = 55
+    push.8589934592
+    exec.accumulate_for_odd_index
+
+    # for j = 56
+    push.1099511627520
+    exec.accumulate_for_even_index
+
+    # for j = 57
+    push.18446744069414551553
+    exec.accumulate_for_odd_index
+
+    # for j = 58
+    push.18428729670905102337
+    exec.accumulate_for_even_index
+
+    # for j = 59
+    push.16140901060737761281
+    exec.accumulate_for_odd_index
+
+    # for j = 60
+    push.68719476736
+    exec.accumulate_for_even_index
+
+    # for j = 61
+    push.8796093020160
+    exec.accumulate_for_odd_index
+
+    # for j = 62
+    push.18446744069414322177
+    exec.accumulate_for_even_index
+
+    # for j = 63
+    push.18302628881338728449
+    exec.accumulate_for_odd_index
+
+    # compute (τ^64 - 1)/ 64
+    dup.3
+    dup.3
+
+    repeat.6
+        dup.1
+        dup.1
+        ext2mul
+    end
+
+    swap
+    sub.1
+    swap
+
+    push.18158513693329981441.0
+    ext2mul
+    ext2mul
+end
+
+#! Given τ and end memory address of interpolated polynomial p ( with degree at max 7, over 
+#! quadratic extension field of Fq ) on stack top, this routine computes α, following formula described
+#! in https://github.com/0xPolygonMiden/miden-vm/issues/592#issue-1481983459
+#!
+#! Expected stack state
+#!
+#! [_, _, τ1, τ0, mem_ptr_p, ...]
+#!
+#! Final stack state
+#!
+#! [α1, α0, _, _, ...]
+#!
+#! Note, mem_ptr_p is absolute memory address of polynomial p ( which was interpolated using advice provider )
+#! s.t. next 3 decreasing ( by 1 ) memory addresses hold remaining 6 coefficients of p. Why this descending order of
+#! memory addressing ? - Read point 2 of https://github.com/0xPolygonMiden/miden-vm/issues/592#issue-1481983459
+proc.compute_alpha
+    movup.4
+    dup
+    sub.1
+    movdn.5
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    dup
+    sub.1
+    movdn.7
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    dup
+    sub.1
+    movdn.7
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    movup.7
+    movup.7
+    ext2mul
+    ext2add
+end
+
+#! Given remainder codeword with 64 evaluations ( as starting memory address on stack ), this routine attempts to
+#! probabilistically verify remainder following the technique described in https://github.com/0xPolygonMiden/miden-vm/issues/592
+#! and an optimization of how to efficiently compute β in https://github.com/0xPolygonMiden/miden-vm/pull/644#pullrequestreview-1265141094
+#!
+#! Expected stack state
+#!
+#! [mem_ptr, ...]
```

Comment:
There is a bit of inconsistency in naming things: we say `mem_ptr` here - but later on we say `mem_ptr_q` and `mem_ptr_p`. It would be good to have the same naming everywhere.

Also, as a nit: it should be OK to say just `ptr` (without `mem`) as pointer usually implies memory pointer. So, I'd do something like `p_ptr` and `q_ptr` - but that's just my preference.

Reply:
Now I've described what `mem_ptr_{p, q}` means.